test/lib/common.js

```diff
@@ -0,0 +1,442 @@
+        }
+
+        done(err, null, res, opts, t, callback);
+        return;
+    }
+
+    if (ifErr(t, err, type + desc)) {
+        done(err, null, res, opts, t, callback);
+        return;
+    }
+
+    t.ok([200, 201].indexOf(res.statusCode) !== -1, 'status code ' + desc);
+
+    if (opts.hasOwnProperty('idKey')) {
+        t.ok(true, fmt('created %s "%s"', opts.type, obj[opts.idKey]));
+    }
+
+    if (opts.exp) {
+        // For creates, the server will generate an ID (usually a UUID) if
+        // it's not set in the request.  Copy this over to the expected
+        // object so that we don't have to set it manually:
+        if (opts.hasOwnProperty('idKey') &&
+                !opts.exp.hasOwnProperty(opts.idKey)) {
+            opts.exp[opts.idKey] = obj[opts.idKey];
+        }
+
+        // Allow filling in values that might be generated before doing the
+        // deepEqual below:
+        if (opts.hasOwnProperty('fillIn')) {
+            opts.fillIn.forEach(function (prop) {
+                if (!opts.exp.hasOwnProperty(prop) &&
+                    obj.hasOwnProperty(prop)) {
+                    opts.exp[prop] = obj[prop];
+                }
+            });
+        }
+
+        var actual = obj;
+        var expected = opts.exp;
+
+        if (opts.hasOwnProperty('ignore')) {
+            var objClone = clone(obj);
+            var expClone = clone(opts.exp);
+
+            opts.ignore.forEach(function (ign) {
+                delete objClone[ign];
+                delete expClone[ign];
+            });
+
+            actual = objClone;
+            expected = expClone;
+        }
+
+        t.deepEqual(actual, expected, type + 'full result' + desc);
+    }
+
+    if (opts.partialExp) {
+        var partialRes = {};
+        for (var p in opts.partialExp) {
+            partialRes[p] = obj[p];
+        }
+
+        t.deepEqual(partialRes, opts.partialExp,
+            type + 'partial result' + desc);
+    }
+
+    if (opts.reqType === 'create') {
+        // We take plural names elsewhere, but expect the singular here:
+        assert.notEqual('s', opts.type.slice(-1));
+        addToState(opts, opts.type + 's', obj);
+    }
+
+    done(null, obj, res, opts, t, callback);
+}
+
+/**
+ * Shared test code for after API delete methods are called
+ */
+function afterAPIdelete(t, opts, callback, err, res) {
+    var desc = opts.desc ? (' ' + opts.desc) : '';
+    assert.string(opts.type, 'opts.type');
+    assert.string(opts.id, 'opts.id');
+    var type = util.format('delete %s %s: ', opts.type, opts.id);
+
+    if (opts.expErr) {
+        t.ok(err, 'expected error');
+        if (err) {
+            var code = opts.expCode || 422;
+            t.equal(err.statusCode, code, 'status code');
+            t.deepEqual(err.body, opts.expErr, 'error body');
+        }
+
+        done(err, null, res, opts, t, callback);
+        return;
+    }
+
+    // mightNotExist allows for calling mod_whatever.dellAllCreated() when
```

Comment:
Neither I do. This is code copied from NAPI as far as I can tell: https://github.com/joyent/sdc-napi/blob/master/test/lib/common.js#L441-L446

My assumption is that `mightNotExist` can be used to prevent errors being bubbled up when those are ResourceNotFound errors expected to be raised by the removal of some objects created and deleted during tests execution.

Reply:
Note that my idea for the future (as you can see on some of the most recent test files) is to move away from this stuff and just run our testing straight into request/response properties as we do for any other APIs, w/o the need for this whole wrapper.